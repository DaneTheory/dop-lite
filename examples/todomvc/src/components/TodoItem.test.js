import React from 'react'
import renderer from 'react-test-renderer';
import TodoItem from './TodoItem'
import { addTodo, editTodo } from '../actions'
import store from '../store'

let component
let idtodo = 1
let indextodo = 0
test('Adding a new todo', () => {  
  // Adding todo manually
  const labelnewtodo = 'My new todo'
  addTodo(labelnewtodo)

  component = renderer.create(
    <TodoItem index={indextodo} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  let labelcomponent = tree.children[0].children[1]
  expect(labelcomponent.type).toBe('label')
  expect(labelcomponent.children[0]).toBe(labelnewtodo)
});


test('Editing todo', () => {
  const labelnewtodo = 'Label changed'
  editTodo(idtodo, 'Label changed')
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  let labelcomponent = tree.children[0].children[1]
  expect(labelcomponent.children[0]).toBe(labelnewtodo)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});