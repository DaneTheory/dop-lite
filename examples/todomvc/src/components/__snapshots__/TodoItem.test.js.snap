// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a new todo 1`] = `
<li
  className=""
>
  <div
    className="view"
  >
    <input
      checked={false}
      className="toggle"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      onDoubleClick={[Function]}
    >
      My new todo
    </label>
    <button
      className="destroy"
      onClick={[Function]}
    />
  </div>
</li>
`;

exports[`Editing todo 1`] = `
<li
  className=""
>
  <div
    className="view"
  >
    <input
      checked={false}
      className="toggle"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      onDoubleClick={[Function]}
    >
      Label changed
    </label>
    <button
      className="destroy"
      onClick={[Function]}
    />
  </div>
</li>
`;

exports[`Editing todo 2`] = `
<li
  className=""
>
  <div
    className="view"
  >
    <input
      checked={false}
      className="toggle"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      onDoubleClick={[Function]}
    >
      Label changed
    </label>
    <button
      className="destroy"
      onClick={[Function]}
    />
  </div>
</li>
`;
